fix: Implement resend confirmation feature for login page

Chesars opened the pull request:
# Pull Request Template

## Description

This PR fixes the non-functional resend confirmation feature on the V3 login page where clicking "Resend confirmation" did nothing. The issue was caused by the V3 store not having the `resendConfirmation` action that the login page was trying to dispatch.

**Key improvements:**
- Fixed V3 store integration by importing `resendConfirmation` directly from auth API
- Added comprehensive UX improvements with loading states and 60-second cooldown timer
- Implemented environment-aware debug logging for development
- Added proper error handling and user feedback
- Enhanced backend test coverage

**Context:** Users with unconfirmed accounts were unable to resend confirmation emails from the login page, creating a poor user experience and potential support burden.

Fixes #3157

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

**Backend Testing:**
- All existing resend_confirmation tests passing (7/7)
- Added comprehensive new test suite in `spec/requests/api/v1/resend_confirmation_spec.rb`
- API endpoint returns 200 OK responses in ~0.39 seconds
- Email delivery confirmed via SMTP with test user `info@airbonar.com`

**Frontend Testing:**
- All frontend tests passing 
- ESLint compliant code with automatic corrections applied
- Manual testing of login page functionality:
  - 60-second cooldown timer with countdown display
  - Error handling with user-friendly messages
  - Development logging works (console output in dev mode only)

**Test Configuration:**
- Ruby/Rails backend with RSpec test suite
- Vue.js frontend with Jest/testing-library
- Development environment with Gmail SMTP configured
- Test user: unconfirmed account `info@airbonar.com`

**Reproduction Steps:**
1. Navigate to login page with unconfirmed account
2. Click "Resend confirmation link"
3. Observe loading state, API call, and success feedback
4. Verify 60-second cooldown prevents spam
5. Check email delivery.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

Chesars committed:
fix: Implement resend confirmation feature for login page

- Fix frontend issue where resend confirmation wasn't working on V3 login page
- Add comprehensive UX improvements with loading states and cooldown timer
- Implement environment-aware debug logging for development
- Add proper error handling and user feedback

Backend improvements:
- Add comprehensive test coverage for resend_confirmation endpoint
- Ensure proper error handling for edge cases

Frontend improvements:
- Fix V3 store integration by importing resendConfirmation directly from auth API
- Add loading state with 'Sending...' indicator during API calls
- Implement 60-second cooldown timer with countdown display
- Add professional yellow banner for unconfirmed accounts
- Add development-only debug logging with structured output
- Proper memory management with cleanup in beforeUnmount lifecycle
- Enhanced error handling with user-friendly messages

Testing:
- All backend tests passing (7/7 resend_confirmation tests)
- All frontend tests passing (2249/2249)
- ESLint compliant code
- API performance validated (200 OK in ~0.39s)
- Email delivery confirmed via SMTP

Fixes issue where resend confirmation link was non-functional on login page.

Chesars committed:
fix: Address RuboCop issues and improve test accuracy

- Use User.from_email for case-insensitive email lookups (RuboCop fix)
- Fix test context naming from 'happy path' to 'when resending confirmation'
- Correct test expectation for non-existent email to return 200 (prevents email enumeration)
- Use be_confirmed matcher in controller tests (RuboCop preference)

All tests now passing with improved code quality and security.

Chesars committed:
fix: Resolve final RuboCop issues in test file

- Replace $alfred.with(&:flushdb) with Redis::Alfred.delete for specific key cleanup
- Replace allow_any_instance_of with class-level allow for cleaner mocking
- Use update! instead of update_column to respect validations
- All RuboCop issues now resolved, tests still passing

Chesars committed:
Merge branch 'develop' into fix/3157-send-confirmation-email-when-trying-to-log-in-if-the-account-it-not-confirmed

Chesars commented:
Hi @sojan-official , please let me know whether this implementation is a high priority or if it doesn’t require much attention yet. I’m happy to keep contributing and learning to help make Chatwoot a better app.

iamsivin commented:
@Chesars Could you please update the PR by resolving the conflicts?

iamsivin commented:
@codex review

Chesars committed:
Merge branch 'develop' into fix/3157-send-confirmation-email-when-trying-to-log-in-if-the-account-it-not-confirmed

Chesars committed:
chore: add i18n translations for resend confirmation feature

Add missing translation keys for the email confirmation resend
functionality in the login page.

iamsivin committed:
Merge branch 'develop' into fix/3157-send-confirmation-email-when-trying-to-log-in-if-the-account-it-not-confirmed

sojan-official committed:
Merge branch 'develop' into fix/3157-send-confirmation-email-when-trying-to-log-in-if-the-account-it-not-confirmed

sony-mathew committed:
fix: repair login resend confirmation frontend

sony-mathew commented on app/javascript/v3/views/login/Index.vue:
```diff
@@ -203,6 +242,82 @@ export default {
+          this.resendConfirmation.cooldownInterval = null;
+        }
+      }, 1000);
     handleMfaVerified() {
```
Fixed with  [20518db](https://github.com/chatwoot/chatwoot/pull/11970/commits/20518db2f5797bdb4fa81610cea824b3a2473b9e)

sony-mathew commented on app/javascript/v3/api/auth.js:
```diff
@@ -101,3 +101,11 @@ export const setNewPassword = async ({
 export const resetPassword = async ({ email }) =>
   wootAPI.post('auth/password', { email });
+
+export const resendConfirmation = async ({ email }) => {
```
Fixed with [20518db](https://github.com/chatwoot/chatwoot/pull/11970/commits/20518db2f5797bdb4fa81610cea824b3a2473b9e)

sony-mathew committed:
Merge branch 'develop' into fix/3157-send-confirmation-email-when-trying-to-log-in-if-the-account-it-not-confirmed

sony-mathew committed:
fix(auth): use unconfirmed login error code

Return a stable user_not_confirmed code for unconfirmed login attempts and preserve it in the v3 login client. Use that code to show the resend confirmation UI instead of matching translated error text, making the flow language agnostic.

sony-mathew commented:
Hey @Chesars, planning to take another direction here. We already have a resend confirmation page. The problem is that it shows up only once after your initial signup. After that when you try to sign-in using unconfirmed email it does not show up.

This is how it looks:
<img width="1026" height="615" alt="Screenshot 2026-05-07 at 2 24 49 PM" src="https://github.com/user-attachments/assets/647ec529-6793-4763-a4ea-8b266bd0fe9f" />



This already has captcha support and rate limiting and all other goodness. If we can re-use this, that would be ideal. 
Being said that we understand that you have put in effort on the current changes.  I will try to salvage this PR as much as I can.

cc: @sojan-official

sony-mathew committed:
fix(auth): route unconfirmed users to verify email

Redirect unconfirmed login attempts to the existing verify-email screen instead of rendering a separate resend confirmation UI on the login form. This keeps resend confirmation on the existing hCaptcha-gated public endpoint and avoids matching locale-dependent login copy in the frontend.

sony-mathew commented on app/javascript/v3/views/login/Index.vue:
```diff
@@ -203,6 +242,83 @@ export default {
+      try {
+        this.logDebug('🚀 Making API call...');
+
+        const response = await resendConfirmation({ email: emailParam });
```
This is not valid now as the approach has changed. Check this: https://github.com/chatwoot/chatwoot/pull/11970#issuecomment-4395717536

sony-mathew commented on app/javascript/v3/views/login/Index.vue:
```diff
@@ -99,6 +104,35 @@ export default {
+        this.loginApi.hasErrored &&
+        this.loginApi.message &&
+        this.loginApi.message.toLowerCase().includes('confirmation')
+      );
```
This is not valid now as the approach has changed. Check this: https://github.com/chatwoot/chatwoot/pull/11970#issuecomment-4395717536

sony-mathew commented on app/javascript/v3/views/login/Index.vue:
```diff
@@ -99,6 +104,35 @@ export default {
+    },
+    resendLinkStyle() {
+      return {
+        color: 'rgb(21, 116, 231)',
```
This is not valid now as the approach has changed. Check this: https://github.com/chatwoot/chatwoot/pull/11970#issuecomment-4395717536

sony-mathew commented:
The latest behaviour is this:

https://github.com/user-attachments/assets/b470a149-7cb0-4883-a725-af8f9d74237d

sony-mathew committed:
Merge branch 'develop' into fix/3157-send-confirmation-email-when-trying-to-log-in-if-the-account-it-not-confirmed

sony-mathew reviewed (approved):
LGTM

Chesars commented:
Hi @sony-mathew 👋 Sounds great, thanks!
